### ISSUE TITLE
MITRE Att&ck integration with STRIDE threats

### ISSUE THREAD

**Issue by saurabhlime** [OWASP/threat-dragon#361]
Describe what problem your feature request solves:
Currently the tool does a good job for decomposing the architecture and applying threats according to the STRIDE framework. This still leaves some gaps to determine the cybersecurity requirements that the process needs to adhere to remediate identified threats.

Describe the solution you'd like:
The solution I am proposing is that we can create an additional layer of taxonomy of threats under STRIDE to identify the attackers tactics and techniques which might be used to identify the right defenses. These can become the cybersecurity requirements for the process that we are threat modelling.

This combined process for threat modeling can be like:
The first step is to identify the process, map out the dataflows and interactions between them and the trust boundaries.[Threat dragon is capable of this]

Second, for each of the subsystems, enumerate a STRIDE matrix listing the mnemonics. 
Third, the 12 ATT&CK tactics are tallied. Enumerated tactics are:
• Initial Access
• Execution
• Persistence
• Privilege Escalation
• Defense Evasion
• Credential Access
• Discovery
• Lateral Movement
• Collection
• Command and Control
• Exfiltration
• Impact

In Step 4, for each of the tactics within each of the STRIDE mnemonics, the applicable techniques are evaluated. For instance, for the STRIDE mnemonic of spoofing, the 12 tactics are evaluated for ATT&CK threat techniques that could result in spoofing against authenticity. In other words, Steps 2 through 4 are a process of elimination. 

Advantages:
1. Using consistent semantics and vocabulary to communicate threats.
2. Understanding the adversary tactics which helps visualize the defenses to those threats.
3. Use open source framework created by MITRE to educate the development teams for various threats and associated remediations.
4. Identify cybersecurity requirements that help defend against multiple threats. [e.g] Preventing Initial access can remediate other threats that are not directly related.  

References:
https://blog.isc2.org/isc2_blog/2020/02/under-attack-how-mitres-methodology-to-find-threats-and-embed-counter-measures-might-work-in-your-or.html
![image](https://user-images.githubusercontent.com/96544391/152564903-135ece7a-6ea5-4840-907f-77f90c7aeb6c.png)

**Comment by storkinsj** [comment]
A thought on this feature:
I think if we go through the stride threats and add something (2), mentally we've already done 3. It's why we know how to do 2. I do think it's a good idea for early threat modelers and will help teach.

My request (as with the pulldown for STRIDE vs other methods) is to default to whatever choice we make, and that it's not a mandatory step. So if we by default not to select the att&ck tactic, we don't get blocked if it's not selected.  That will allow the "power users" to continue to move quickly through the model.

If you've used the later iteration of the MS SDL threat modeler, you know it's hardwired to windows mechanics and the wizard based approach can break threat modeling by forcing you into choices that don't apply. This is why I've moved away from that tool.

**Comment by saurabhlime** [comment]
I completely agree with not mandating the Att&ck tactic step. Though we can pre-select the att&ck layer based on the process that is selected for Threat modelling(Cloud, container, network, IAAS etc).

**Comment by jgadsden** [comment]
Agreed - where Threat Dragon wins is where it is simple and intuitive to use. The freeform text boxes are a big bonus for our community. If we implement this feature we would need it to be optional, as in it would not intercept the main use case.

From my point of view I would find this very useful, so could it be a different mode of Threat Dragon? I am not sure what I mean by that ... somehow I would be able to switch in to and out of Att&ck integration?

**Comment by saurabhlime** [comment]
Helpful resource to visualize STRIDE mapping to MITRE's Attack CAPEC.
https://ostering.com/blog/2022/03/07/capec-stride-mapping/

**Comment by jgadsden** [comment]
Another comment from a Threat Dragon user: "I am looking to use Threat Dragon as a tool which will inform risk assessments. I am interested to understand if you are aware iof it is possible to import the MITRE CAPEC "Mechanisms of Attack" methods as a .json file? I understand this may be possible as importing it as a layer, and then on top of this doing the actual threat modelling."

**Comment by jgadsden** [comment]
Work on the threat engine is planned for version 2.4 , targeted for end of 2024
This issue should be considered for that milestone bit if not feasible then this issue can be closed

**Comment by mademarc** [comment]
Hey @jgadsden i think the best way to go with this is to enhance the current STRIDE framework, I propose integrating MITRE ATT&CK tactics and techniques into the threat modeling process. This will map each STRIDE threat category to relevant ATT&CK tactics (e.g., Initial Access, Persistence) and techniques (e.g., Phishing, Credential Dumping)

a couple of sotuols would be the following:

1. Map STRIDE threats to ATT&CK tactics: For each STRIDE category (e.g., Spoofing, Tampering), identify corresponding ATT&CK tactics.

2. Evaluate techniques: For each tactic, assess techniques that apply to the STRIDE threat.

And 

3.Cybersecurity requirements: Use this mapping to generate clearer, actionable defense requirements to mitigate multiple related threats.

And that all said here i think this integration will streamline threat modeling, help identify gaps in defenses, and create more comprehensive remediation strategies.

Could i help/ assist with this enhancement by creating the necessary mappings and integrating them into the tool? :)

**Comment by jgadsden** [comment]
Hello @mademarc , certainly what you describe + propose makes a lot of sense and is a good way forward.
It would be great of you could start work on this, I can assign this issue to you
